### PR TITLE
Add known issue for WSL2 and cgroupv2 support

### PR DESCRIPTION
**Pull request by stmcginnis**
This adds a note to the Known Issues document with details related to proper setup requirements for cgroupv2 support and links to the known issues page for extra visibility from the Using WSL2 page.

Includes a separate commit that cleans up some of the stale information from the known issues page and minor formatting issues found while performing updates.

Closes: #3685

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm
/kind documentation

File: site/content/docs/user/known-issues.md [hunk 1]
```diff
@@ -389,37 +375,46 @@ docker: Error response from daemon: open /dev/dma_heap: permission denied.
...
+```
+
+Using KIND in these environments will require upgrading your OS to a more recent version that supports cgroup namespaces.
+Another option is to run a virtual machine using a newer kernel.
+
+## Failure to Create Cluster on WSL2
 
-Docker Desktop 4.3.0+ uses Cgroups v2 (see [release notes](https://docs.docker.com/release-notes/)) and can only work with Kubernetes versions >= 1.19 (see [failure to create cluster with Cgroups v2](https://kind.sigs.k8s.io/docs/user/known-issues/#failure-to-create-cluster-with-cgroups-v2)).
+Some Linux kernel options for WSL2 do not have cgroup configured in a way that
+Kind and other Linux-focused tools may expect. This may result in a failure
```
Comment (BenTheElder) [inline site/content/docs/user/known-issues.md:400]: nit: `kind`  or `KIND`

File: site/content/docs/user/known-issues.md [hunk 2]
```diff
@@ -389,37 +375,46 @@ docker: Error response from daemon: open /dev/dma_heap: permission denied.
...
+unable to start container process: error adding pid 655569 to cgroups
+```
 
-You may also be able to make a configuration change to your Docker Desktop
-settings. Starting with Docker Desktop 4.4.2 a deprecated option has been added
-to support those requiring cgroupv1. As noted in the [4.4.2 release
-notes](https://docs.docker.com/desktop/mac/release-notes/#docker-desktop-442),
-the setting `deprecatedCgroupv1` can be set to `true` in `settings.json`. After
-restarting the Docker Engine, the VM used by Docker Desktop will use cgroupv1.
+The Kind development team is not able to provide support with Windows and WSL, so
```
Comment (BenTheElder) [inline site/content/docs/user/known-issues.md:407]: nit again, we don't use `Kind` in the rest of the docs.

File: site/content/docs/user/known-issues.md [hunk 3]
```diff
@@ -389,37 +375,46 @@ docker: Error response from daemon: open /dev/dma_heap: permission denied.
 
 Although the policy has been fixed in Fedora 34, the fix has not been backported to Fedora 33 as of June 28, 2021. Putting SELinux in permissive mode (`setenforce 0`) is one known workaround. This disables SELinux until the next boot. For more details, see [kind#2296].
 
-## Failure to Create Cluster with Docker Desktop as Container Runtime
+## Docker Desktop for macOS and Windows
+
+Docker containers cannot be executed natively on macOS and Windows, therefore Docker Desktop runs them in a Linux VM. As a consequence, the container networks are not exposed to the host and you cannot reach the kind nodes via IP. You can work around this limitation by configuring [extra port mappings](https://kind.sigs.k8s.io/docs/user/configuration/#extra-port-mappings) though.
```
Comment (BenTheElder) [inline site/content/docs/user/known-issues.md:380]: This isn't the only solution and we should elaborate more if we're going to mention it here, they have a number of options including the recent work in cloud provider kind, proxies, running tests inside the cluster network, etc.

Maybe at least "You can work around this limitation by, for example, configuring [extra port mappings] amongst other options"

File: site/content/docs/user/known-issues.md [hunk 4]
```diff
@@ -389,37 +375,46 @@ docker: Error response from daemon: open /dev/dma_heap: permission denied.
...
+
+## Older Linux Distributions
+
+KIND uses a cgroup setting of `cgroupns=private`. The cgroup namespace functionality was added in 2016, so some of the
```
Comment (BenTheElder) [inline site/content/docs/user/known-issues.md:384]: (which is on by default in cgroups v2 on all major container runtimes)

**Review by BenTheElder — COMMENTED** [review]
thanks for working on this, did a first pass

cc @aojea

File: site/content/docs/user/known-issues.md (hunk 3 above)
Comment (stmcginnis) [reply site/content/docs/user/known-issues.md:380]: This is showing up as new in the diff, but this is an existing section: https://kind.sigs.k8s.io/docs/user/known-issues/#docker-desktop-for-macos-and-windows

I'll tweak the wording a bit to try to improve it though.

File: site/content/docs/user/known-issues.md [hunk 5]
```diff
@@ -209,19 +202,12 @@ docker image prune
 
 You can verify the issue by exporting the logs (`kind export logs`) and looking
 at the kubelet logs, which may have something like the following:
-```
+
+```txt
 Dec 07 00:37:53 kind-1-control-plane kubelet[688]: I1207 00:37:53.229561     688 eviction_manager.go:340] eviction manager: must evict pod(s) to reclaim ephemeral-storage
 Dec 07 00:37:53 kind-1-control-plane kubelet[688]: E1207 00:37:53.229638     688 eviction_manager.go:351] eviction manager: eviction thresholds have been met, but no pods are active to evict
 ```
-## Failure to Create Cluster with Cgroups v2
```
Comment (BenTheElder) [inline site/content/docs/user/known-issues.md:216]: I guess we can refer back to this in git / ask people to please upgrade. It's surprising how often we still get issues from really old releases of both kind and Kubernetes

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve
thanks and sorry for the delay

**Pull request merged**